server/src/main/java/org/opensearch/cluster/routing/allocation/DiskThresholdMonitor.java

```diff
@@ -213,6 +220,46 @@ public void onNewInfo(ClusterInfo info) {
                     }
             }
 
+            // Check for file descriptor usage
+            ProcessStats processStats = info.getNodeProcessStats().get(usage.getNodeId());
+            if (processStats != null) {
+                long openFDs = processStats.getOpenFileDescriptors();
+                long maxFDs = processStats.getMaxFileDescriptors();
+
+                if (maxFDs > 0) {
+                    double fdUsagePercent = (openFDs * 100.0) / maxFDs;
+
+                    // Applying write block at 85%
+                    if (fdUsagePercent >= 85.0) {
+                        if (nodesOverFileDescriptorThreshold.add(node)) {
+                            logger.warn(
+                                "file descriptor usage [{}%] exceeded 85% threshold on {}, marking indices read-only",
+                                String.format("%.1f", fdUsagePercent),
+                                usage.getNodeName()
+                            );
+                        }
+                        if (routingNode != null) {
+                            for (ShardRouting routing : routingNode) {
+                                String indexName = routing.index().getName();
+                                indicesToMarkReadOnly.add(indexName);
+                                indicesNotToAutoRelease.add(indexName); // Prevent auto-release
+                            }
+                        }
+                    }
+                    // Releasing write block at 75%
+                    else if (fdUsagePercent <= 75.0
+                        && !nodeDiskEvaluator.isNodeExceedingFloodStageWatermark(usage)
+                        && !nodeDiskEvaluator.isNodeExceedingHighWatermark(usage)
```

Comment:
I am not sure if this is the right way to remove write block. Instead can we move this check below disk watermark and make the check on indicesToMarkReadOnly?